config/config.go

```diff
@@ -294,6 +305,127 @@ func (m *ManualWarpMessage) Validate() error {
 	return nil
 }
 
+// If the numerator in the Warp config is 0, use the default value
+func setQuorumNumerator(cfgNumerator uint64) uint64 {
+	if cfgNumerator == 0 {
+		return warp.WarpDefaultQuorumNumerator
+	}
+	return cfgNumerator
+}
+
+// Helper to retrieve the Warp Quorum from the chain config.
+// Differentiates between subnet-evm and coreth RPC internally
+func getWarpQuorum(
+	blockchainID ids.ID,
+	subnetID ids.ID,
+	client ethclient.Client,
+) (WarpQuorum, error) {
+	if subnetID == constants.PrimaryNetworkID {
+		return WarpQuorum{
+			QuorumNumerator:   warp.WarpDefaultQuorumNumerator,
+			QuorumDenominator: warp.WarpQuorumDenominator,
+		}, nil
+	}
```

Comment:
Warp can't send messages to the P and X chains, can it? Do we know what happens if someone tried to? Is there a way we can grab the blockchainID for the c-chain?

Reply:
> Warp can't send messages to the P and X chains, can it? Do we know what happens if someone tried to?
That's correct for now, but in the future I would expect warp support on all chains, including X and P. There technically isn't way to try to send a message to the X or P chain since their VMs do not support it at all.

I'm neither here or there on using the C-chain blockchain ID as the condition here rather than the primary network subnet ID though.

Reply:
The Warp protocol is compatible with the P and X chains, though support has not been implemented (nor am I aware of any plans to do so).

However, since neither of them are EVM chains, the `evm` VM type in the relayer config will fail to work properly for those chains. I'm not exactly sure what would happen, but it wouldn't result in a tx posted to either of those chains.

> Is there a way we can grab the blockchainID for the c-chain?

There are endpoints to map subnet ID <-> blockchain ID, but the current config requires both to be specified by the user. I think this is the most general approach setup, since no subnet is guaranteed to have a fixed number of blockchains in perpetuity. I also think we should keep the primary network special casing as minimal as possible.

Reply:
Thanks!

Reply:
> I'm neither here or there on using the C-chain blockchain ID as the condition here rather than the primary network subnet ID though.

Using the subnetID would be simpler because the primary network subnet ID is always the constant `ids.Empty`, whereas the C-Chain blockchain ID would have to be fetched from the network.